### PR TITLE
Fix the previous order totals for subsequent amends

### PR DESCRIPTION
**Pull request by martinkimmings**
This PR removes the `TotalCost` extension method for an `order` in favour off the `orderWrapper` methods `TotalCost` and  `TotalPreviousCost`. 

Calls like `OrderWrapper.Previous.TotalCost()` are replaced with `OrderWrapper.TotalPreviousCost()` which carries out the calculation for the original order (using the maximum term) and each revision (using a term derived from the planned delivery dates).

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.Framework/Calculations/CataloguePriceCalculations.cs [hunk 1]
```diff
@@ -101,12 +100,14 @@ public static decimal TotalAnnualCost(this Order order, bool roundResult = false
             return total;
         }
 
-        public static decimal TotalCost(this Order order, bool roundResult = false)
+        public static decimal TotalPreviousCost(this OrderWrapper orderWrapper, bool roundResult = false)
         {
-            var maximumTerm = order?.MaximumTerm ?? 36;
+            if (orderWrapper == null)
+            {
+                return decimal.Zero;
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.Framework/Calculations/CataloguePriceCalculations.cs:107]: For the sake of completeness, let's get a unit test added ot test this edge case.

**Review by bjss-colin — CHANGES_REQUESTED** [review]

**Review by bjss-colin — APPROVED** [review]

**Review by imranbashir01 — APPROVED** [review]
